features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/data/APIProduct.jsx

```diff
@@ -0,0 +1,396 @@
+/**
+ * Copyright (c) 2019, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* eslint-disable */
+import API from 'AppData/api';
+import APIClientFactory from './APIClientFactory';
+import Utils from './Utils';
+import Resource from './Resource';
+
+/**
+ * An abstract representation of an API Product
+ */
+class APIProduct extends Resource {
+
+    constructor(name, version, context, kwargs) {
+        super();
+        let properties = kwargs;
+        if (name instanceof Object) {
+            properties = name;
+        } else {
+            this.name = name;
+            this.version = '1.0.0';
+            this.context = context;
+            this.gatewayEnvironments = ['Production and Sandbox'];
+            this.transport = [
+                'http',
+                'https',
+            ];
+            this.visibility = 'PUBLIC';
+        }
+        this._data = properties;
+        for (const key in properties) {
+            if (Object.prototype.hasOwnProperty.call(properties, key)) {
+                this[key] = properties[key];
+            }
+        }
+        this.apiType = API.CONSTS.APIProduct;
+        this.getType = this.getType.bind(this);
+    }
+
+    /**
+     *
+     * @param data
+     * @returns {object} Metadata for API request
+     * @private
+     */
+    _requestMetaData() {
+        Resource._requestMetaData();
+    }
+
+    getType() {
+        return this.type;
+    }
+
+    /**
+     * Create an API Product with the given parameters in template
+     * @param {Object} api_product_data - APIProduct data which need to fill the placeholder values in the @get_template
+     * @returns {Promise} Promise after creating API Product
+     */
+    create(api_product_data) {
+        let payload;
+        let promise_create;
+        payload = {
+            body: api_product_data,
+            'Content-Type': 'application/json'
+        };
+        promise_create = this.client.then((client) => {
```

Comment:
shall we change these variables to camel case

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/6759/commits/0c218762d38e7db55b9b2b6306a4a1027e6d4cb0